client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -145,7 +150,11 @@ export default function Confirm( props: WooCommerceInstallProps ): ReactElement
 		);
 	}
 
-	if ( ! siteId || ! isDataReady || isAtomicSite || isReadyForTransfer ) {
+	if (
+		! siteId ||
+		! isDataReady ||
+		( ( isAtomicSite || isReadyForTransfer ) && ! siteUpgrading?.required )
```

Comment:
Just double-checking that this logic is correct. It feels like it should be the opposite of the boolean logic that submits the step. Wouldn't it be something like:

```suggestion
		! siteId ||
		! isDataReady ||
		siteUpgrading?.required ||
		! ( isAtomicSite || isReadyForTransfer )
```

Why would you want to show loading ellipsis when the user doesn't need to upgrade?

Reply:
I think I understand now. The `useEffect` hook isn't the only way for this step to be submitted, there's also a button click handler. It looks like `! siteUpgrading?.required` means we're _not_ going to go to the checkout, which means we don't want to show the ellipses on the way to the checkout. Which is what we see in the video you've attached.

Reply:
Yea its a little odd,

The first part blocks auto transfer when there is an upgrade. This is because we've filtered out the blockers that would usually prevent this.

The second part only controls logic about when it should show the loading thing. If it's upgrading required, we still want to show it while its loading, but don't want it to be a trigger for always showing it. Once data is available we want to show the confirm button.